capsules/src/touch.rs

```diff
@@ -203,71 +218,76 @@ impl<'a> hil::touch::MultiTouchClient for Touch<'a> {
         } else {
             num_events
         };
-        // debug!("{} touch(es)", len);
-        for app in self.apps.iter() {
-            app.enter(|app, upcalls| {
-                if app.ack {
-                    app.dropped_events = 0;
+        if len > 0 {
+            // report the first touch as single touch
+            // for applications that only use single touches
+            self.touch_event(touch_events[0]);
```

Comment:
Wouldn't it be odd for a multi touch app to get a single touch callback (as well as the multi touch version)?

Reply:
Apps subscribe to different upcall numbers for touch and multitouch. The touch driver usually has only one of the two underlying drivers: either single touch either multitouch.

Apps that need only one touch would have to check the number of touches and subscribe to different upcalls accordingly. This allows apps that only need a single touch to subscribe to the single touch upcall regardless of what kind of hardware the touch driver uses.

Reply:
Gotcha.